plugins/check_mysql.c

```diff
@@ -58,7 +62,7 @@ char *ciphers = NULL;
 bool ssl = false;
 char *opt_file = NULL;
 char *opt_group = NULL;
-unsigned int db_port = MYSQL_PORT;
+unsigned int db_port = CHECK_PORT_DEFAULT;
```

Comment:
My suggestion here would be
#ifndef MYSQL_PORT
#define MYSQL_PORT 3306
#endif 
to retain the same behaviour on old installs and systems with MySQL, instead of MariaDB.

Reply:
But why go out of our way to do the wrong thing sometimes, when we could do the right thing unconditionally?

Maybe I'm asking the wrong question... let's make sure we're on the same foot: does anyone think that "whatever the default MySQL port was on the machine where monitoring-plugins was compiled" is a useful default for end-users of monitoring-plugins? (Consider that most users of monitoring-plugins have no idea where it was compiled.)

Reply:
For distributions that do ship packages in binary (eg. Fedora, CentOS, RHEL, SLES), it shouldn't matter, since they do compile 'em in an isolated environment.
That's why I'd suggest to use ifndef, since if someone really _intentionally_ want's to compile the check plugin differently, it's easier to define MYSQL_PORT somewhere, instead of changing the source and use a different _static_ value.

Reply:
Is it easier to add `#define MYSQL_PORT 8675309` than it is to edit the existing line to say `#define CHECK_PORT_DEFAULT 8675309` instead of `#define CHECK_PORT_DEFAULT 3306`? I think they're roughly the same difficulty.

And besides, there are exactly zero people who want to do that. If you need a specific port, you can specify it on the command-line.

With all else being equal, the simpler shorter code that acts predictably is more fun to maintain.

Reply:
Sure it's the same "difficulty", no matter if it's called MYSQL_PORT or CHECK_PORT_DEFAULT. My intention/idea was to keep the same behaviour as it was (use port from MariaDB/MySQL header files), but allow override (eg. another MySQL fork comes along and uses another port...). And I'm a huge fan of reusing existing static values, if reasonable.
But it's your source and for sure, defining your own static value will probably increase predictability.

Reply:
Ping also from my side. Is there any intention to make this work for everybody?